perfkitbenchmarker/providers/ibmcloud/util.py

```diff
@@ -0,0 +1,212 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utilities for working with IBM Cloud resources."""
+
+import os
+import json
+import yaml
+
+from perfkitbenchmarker import errors
+from perfkitbenchmarker.providers.ibmcloud import ibmcloud
+
+WINDOWS = 'windows'  # all windows
+UBUNTU = 'ubuntu'  # all ubuntu
+REDHAT = 'redhat'  # all redhat
+DEBIAN = 'debian'  # all debian
+CENTOS = 'centos'  # all centos
+UNKNOWN = 'unknown'
+
+SUBNET_SUFFIX = 's'
+DELIMITER = 'z-z'
+
+# these are used to create extra subnets for multi vnic support in perfkit,
+# not used for vm provisioning, up to 4 extra totaling 5 subnets
+SUBNET_SUFFIX_EXTRA = 'sxs'
+SUBNETX1 = SUBNET_SUFFIX_EXTRA + '1'
+SUBNETX2 = SUBNET_SUFFIX_EXTRA + '2'
+SUBNETX3 = SUBNET_SUFFIX_EXTRA + '3'
+SUBNETX4 = SUBNET_SUFFIX_EXTRA + '4'
+SUBNETXS = [SUBNETX1, SUBNETX2, SUBNETX3, SUBNETX4]
+
+SUBNETS_EXTRA = {
+    SUBNETX1: ['10.101.20.0/24', '10.102.20.0/24', '10.103.20.0/24', '10.104.20.0/24', '10.105.20.0/24'],
+    SUBNETX2: ['10.101.30.0/24', '10.102.30.0/24', '10.103.30.0/24', '10.104.30.0/24', '10.105.30.0/24'],
+    SUBNETX3: ['10.101.40.0/24', '10.102.40.0/24', '10.103.40.0/24', '10.104.40.0/24', '10.105.40.0/24'],
+    SUBNETX4: ['10.101.50.0/24', '10.102.50.0/24', '10.103.50.0/24', '10.104.50.0/24', '10.105.50.0/24']
+}
+
+SUBNETS_EXTRA_GATEWAY = {
+    SUBNETX1: ['10.101.20.1', '10.102.20.1', '10.103.20.1', '10.104.20.1', '10.105.20.1'],
+    SUBNETX2: ['10.101.30.1', '10.102.30.1', '10.103.30.1', '10.104.30.1', '10.105.30.1'],
+    SUBNETX3: ['10.101.40.1', '10.102.40.1', '10.103.40.1', '10.104.40.1', '10.105.40.1'],
+    SUBNETX4: ['10.101.50.1', '10.102.50.1', '10.103.50.1', '10.104.50.1', '10.105.50.1']
+}
+
+
+def ReadConfig(config):
+  """Reads in config yml
+
+  Args:
+    config: config file name.
+
+  Returns:
+    The json representation of the config file.
+  """
+  try:
+    with open(config, 'r') as stream:
+      data = json.dumps(yaml.load(stream), sort_keys=True)
+      return json.loads(data)
+  except Exception as ex:
+    raise errors.Error('Failed to load configuration file %s, %s', config, ex)
+
+
+def GetSubnetIndex(ipv4_cidr_block):
+  """Finds the index for the given cidr
+
+  Args:
+    ipv4_cidr_block: cidr to find.
+
+  Returns:
+    The index number of the found cidr as in the predefined list
+  """
+  for name in SUBNETXS:
+    ip_list = SUBNETS_EXTRA[name]
+    for i in range(len(ip_list)):
+      if ip_list[i] == ipv4_cidr_block:
+        return i
+  return -1
```

Comment:
Might want to consider throwing an exception here if the caller expects to get a subnet.  If it is optional then note in the Returns that -1 is returned for missing.

Reply:
I just noted it returns -1 for missing.